packages/phpstan-rules/src/Rules/RequireStringRegexMatchKeyRule.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\PHPStanRules\Rules;
+
+use Nette\Utils\Strings;
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\ArrayDimFetch;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Scalar\LNumber;
+use PhpParser\NodeFinder;
+use PHPStan\Analyser\Scope;
+use Symplify\PHPStanRules\Printer\NodeComparator;
+use Symplify\PHPStanRules\ValueObject\PHPStanAttributeKey;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Symplify\PHPStanRules\Tests\Rules\RequireStringRegexMatchKeyRule\RequireStringRegexMatchKeyRuleTest
+ */
+final class RequireStringRegexMatchKeyRule extends AbstractSymplifyRule
+{
+    /**
+     * @var string
+     */
+    public const ERROR_MESSAGE = '"%s" regex need to use string named capture group instead of numeric';
+
+    /**
+     * @var NodeFinder
+     */
+    private $nodeFinder;
+
+    /**
+     * @var NodeComparator
+     */
+    private $nodeComparator;
+
+    public function __construct(NodeFinder $nodeFinder, NodeComparator $nodeComparator)
+    {
+        $this->nodeFinder = $nodeFinder;
+        $this->nodeComparator = $nodeComparator;
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Assign::class];
+    }
+
+    /**
+     * @param Assign $node
+     * @return string[]
+     */
+    public function process(Node $node, Scope $scope): array
+    {
+        if ($this->shouldSkipExpr($node->expr)) {
+            return [];
+        }
+
+        /** @var Node|null $parent */
+        $parent = $node->getAttribute(PHPStanAttributeKey::PARENT);
```

Comment:
Looking only for first parent would miss nested calls like:

```php
if ($matches = Strings::match(...)) {
    echo $matches[0] 
}
```

Here should be something like:

```php
$parent = $this->getFirstParent($node, [FunctionLike::class]);

```

Reply:
there is already fixture for that : NumericDimInsideIfCond.php

Reply:
Ok, let's give it a go